### PR TITLE
Develop team pick pack ship entities

### PR DESCRIPTION
**Pull request by ian-a-hickey**
Opening this running pull request for daily review and touchbase.

---
This change is [Reviewable](https://reviewable.io/reviews/ten24/slatwall/5312)

### CODE REVIEW

**Comment by rmarchandTen24** [comment]
need to address many-to-many linktable being over 30 characters. need to update tests to give exact feedback on that

File: model/entity/FulfillmentBatchItem.cfc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	property name="createdByAccountID" hb_populateEnabled="false" ormtype="string";
+	property name="modifiedDateTime" hb_populateEnabled="false" ormtype="timestamp";
+	property name="modifiedByAccountID" hb_populateEnabled="false" ormtype="string";
+
+	// ============ START: Non-Persistent Property Methods =================
+
+	// ============  END:  Non-Persistent Property Methods =================
+
+	// ============= START: Bidirectional Helper Methods ===================
+
```
Comment (rmarchandTen24) [inline model/entity/FulfillmentBatchItem.cfc]: Need bidirectionals

File: model/entity/PickWave.cfc [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	// ============ START: Non-Persistent Property Methods =================
+
+	// ============  END:  Non-Persistent Property Methods =================
+
+	// ============= START: Bidirectional Helper Methods ===================
+	
+	// Fulfillment Batch Items (one-to-many)
+	public void function addFulfillmentBatchItem(required any fulfillmentBatchItem) {
+		arguments.fulfillmentBatchItem.setPickWave( this );
```
Comment (rmarchandTen24) [inline model/entity/PickWave.cfc]: Can you create the bi-directionals on the other side

File: model/process/FulfillmentBatch_Create.cfc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    If you modify this program, you may extend this exception to your version 
+    of the program, but you are not obligated to do so.
+
+Notes:
+
+*/
+component output="false" accessors="true" extends="HibachiProcess" {
+
+	// Injected Entities
+	property name="fulfillmentBatch";
```
Comment (rmarchandTen24) [inline model/process/FulfillmentBatch_Create.cfc]: Can you add the meta data for this as an object. It will help with the angular side

**Comment by ian-a-hickey** [comment]
I shortened the link table name that was an issue. 
I updated all the bidirectionals to have both sides of the relationship.
I added some helper methods to fulfillmentBatch to make it easy to get some status data.

File: model/entity/Account.cfc [hunk 4]
```diff
@@ -96,7 +96,9 @@ component displayname="Account" entityname="SlatwallAccount" table="SwAccount" p
 	property name="subscriptionUsages" singularname="subscriptionUsage" cfc="SubscriptionUsage" type="array" fieldtype="one-to-many" fkcolumn="accountID" cascade="all-delete-orphan" inverse="true";
 	property name="termAccountOrderPayments" singularname="termAccountOrderPayment" cfc="OrderPayment" type="array" fieldtype="one-to-many" fkcolumn="termPaymentAccountID" cascade="all" inverse="true";
 	property name="giftCards" singularname="giftCard" cfc="GiftCard" type="array" fieldtype="one-to-many" fkcolumn="ownerAccountID" cascade="all" inverse="true";
-
+	property name="fulfillmentBatches" singularname="fulfillmentBatch" fieldType="one-to-many" type="array" fkColumn="accountID" cfc="fulfillmentBatch" inverse="true" orderby="createdDateTime asc";
```
Comment (rmarchandTen24) [inline model/entity/Account.cfc]: CFC needs to be capitalized

Comment (rmarchandTen24) [inline model/entity/Account.cfc]: Please remove the orderby. We enforce that via smart list.

File: model/entity/OrderDeliveryItem.cfc [hunk 5]
```diff
@@ -59,7 +59,8 @@ component displayname="Order Delivery Item" entityname="SlatwallOrderDeliveryIte
 
 	// Related Object Properties (one-to-many)
 	property name="referencingOrderItems" singularname="referencingOrderItem" cfc="OrderItem" fieldtype="one-to-many" fkcolumn="referencedOrderDeliveryItemID" inverse="true" cascade="all"; // Used For Returns
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="orderDeliveryItemID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
Comment (rmarchandTen24) [inline model/entity/OrderDeliveryItem.cfc]: Capitalize cfc

Comment (rmarchandTen24) [inline model/entity/OrderDeliveryItem.cfc]: Remove order y

File: model/entity/OrderItem.cfc [hunk 6]
```diff
@@ -78,7 +78,8 @@ component entityname="SlatwallOrderItem" table="SwOrderItem" persistent="true" a
 	property name="accountLoyaltyTransactions" singularname="accountLoyaltyTransaction" cfc="AccountLoyaltyTransaction" type="array" fieldtype="one-to-many" fkcolumn="orderItemID" cascade="all" inverse="true";
 	property name="giftCards" singularname="giftCard" cfc="GiftCard" type="array" fieldtype="one-to-many" fkcolumn="originalOrderItemID" cascade="all" inverse="true";
 	property name="orderItemGiftRecipients" singularname="orderItemGiftRecipient" cfc="OrderItemGiftRecipient" type="array" fieldtype="one-to-many" fkcolumn="orderItemID" cascade="all" inverse="true";
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="orderItemID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
Comment (rmarchandTen24) [inline model/entity/OrderItem.cfc]: CFC capitalize. Remove orderby

File: model/entity/Stock.cfc [hunk 7]
```diff
@@ -58,7 +58,8 @@ component displayname="Stock" entityname="SlatwallStock" table="SwStock" persist
 	// Related Object Properties (one-to-many). Including this property to allow HQL to do  stock -> vendorOrderItem lookups
 	property name="vendorOrderItems" singularname="vendorOrderItem" cfc="VendorOrderItem" fieldtype="one-to-many" fkcolumn="stockID" inverse="true";
 	property name="inventory" singularname="inventory" cfc="Inventory" fieldtype="one-to-many" fkcolumn="stockID" inverse="true" lazy="extra";
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="stockID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
Comment (rmarchandTen24) [inline model/entity/Stock.cfc]: CFC capitalize. Remove orderby

File: model/entity/Account.cfc [hunk 8]
```diff
@@ -96,7 +96,9 @@ component displayname="Account" entityname="SlatwallAccount" table="SwAccount" p
...
+	property name="pickWaves" singularname="pickWave" fieldType="one-to-many" type="array" fkColumn="accountID" cfc="PickWave" inverse="true" orderby="createdDateTime asc";
```
Comment (rmarchandTen24) [inline model/entity/Account.cfc]: Remove orderby

File: model/entity/OrderFulfillment.cfc [hunk 9]
```diff
@@ -73,7 +73,8 @@ component displayname="Order Fulfillment" entityname="SlatwallOrderFulfillment"
 	property name="fulfillmentShippingMethodOptions" singularname="fulfillmentShippingMethodOption" cfc="ShippingMethodOption" fieldtype="one-to-many" fkcolumn="orderFulfillmentID" cascade="all-delete-orphan" inverse="true";
 	property name="accountLoyaltyTransactions" singularname="accountLoyaltyTransaction" cfc="AccountLoyaltyTransaction" type="array" fieldtype="one-to-many" fkcolumn="orderFulfillmentID" cascade="all" inverse="true";
 	property name="attributeValues" singularname="attributeValue" cfc="AttributeValue" type="array" fieldtype="one-to-many" fkcolumn="orderFulfillmentID" cascade="all-delete-orphan" inverse="true";
-
+	property name="fulfillmentBatchItems" singularname="fulfillmentBatchItem" fieldType="one-to-many" type="array" fkColumn="orderFulfillmentID" cfc="fulfillmentBatchItem" inverse="true" orderby="createdDateTime asc";
```
Comment (rmarchandTen24) [inline model/entity/OrderFulfillment.cfc]: Capitalize cfc. Remove orderby

File: model/entity/Account.cfc (hunk 4 above)
Comment (ian-a-hickey) [inline model/entity/Account.cfc]: removed the orderBy and capitalized

**Pull request closed without merging**